### ISSUE TITLE
[CLASSIC] Jeklik Encounter Zul'Gurub

### ISSUE THREAD

**Issue by kamicolo** [cmangos/issues#839]
## Source:

(cmangos-dev / src / scriptdev2 / scripts / eastern_kingdoms / zulgurub / boss_jeklik.cpp)
SD%Complete: 85
Jeklik Research:
## Inside script currently:

35     // Bat spells 
36     SPELL_CHARGE                = 22911, 
37     SPELL_SONIC_BURST           = 23918, 
38     // SPELL_PSYHIC_SCREAM       = 22884,                   // spell not confirmed - needs research 
39     SPELL_SWOOP                 = 23919, 
40     SPELL_SUMMON_FRENZIED_BATS  = 23974, 
41  
42     // Troll form spells 
43     SPELL_SHADOW_WORD_PAIN      = 23952, 
44     SPELL_MIND_FLAY             = 23953, 
45     SPELL_BLOOD_LEECH           = 22644, 
46     SPELL_GREATERHEAL           = 23954, 
47  
48     // Common spells 
49     SPELL_GREEN_CHANNELING      = 13540,                    // visual for idle mode 
50     SPELL_BAT_FORM              = 23966, 
51  
52     // Batriders Spell 
53     SPELL_LIQUID_FIRE           = 23968,                    // script effect - triggers 23971, 
54     SPELL_UNSTABLE_CONCOCTION   = 24024, 
55     SPELL_TRASH                 = 8876, 
56     SPELL_DEMORALIZING_SHOUT    = 23511, 
57     SPELL_BATTLE_COMMAND        = 5115, 
58     SPELL_INFECTED_BITE         = 16128, 

Going to post each spell on a differnt note to keep everything seperate and as structured and tidy possible.

**Comment by kamicolo** [comment]
Additional note:
source: http://ru-mangos.ru/files/earlyServers/1.8.0/scripts/2006-04-19-Some%20Fixes%20for%20Boss%20And%20Instance%20Mob%20Spells.txt

namespace eval ::ai_HighPriestessJeklik {
	proc CanCast { npc victim } {
		set mobh [ ::GetHealthPCT $npc ]
		set spell_list "24437 23919 24673 12097 24687"
		;# Blood Leech, Swoop, Curse of Blood, Pierce Armor, Aspect of Jeklik
		set transform_spell_list "24209 23952 23953 15398 23966"
		;# Great Heal, Shadow Word: Pain, Mind Flay, Psychic Scream, Transform Troll
		set spellid [ lindex $spell_list [ expr { int( rand() * [ llength $spell_list ] ) } ] ]
		if { $mobh == 50 } {
			set spellid 23966
		} elseif { $mobh < 50 } {
			set spellid [ lindex $transform_spell_list [ expr { int( rand() * [ llength $transform_spell_list ] ) } ] ]
			if { $spellid == 24209 } { return [ ::AI::Cast $npc $npc $spellid ] }
		} else {
			set spellid [ lindex $spell_list [ expr { int( rand() * [ llength $spell_list ] ) } ] ]
		}
		return [ ::AI::Cast $npc $victim $spellid ]
	}
}

Blood leech mention / swoop / curse of blood / pierce armor / aspect of jeklik.

**Comment by kamicolo** [comment]
Additional note:
source: http://www.disciplesguild.com/forums/topic/end-game-instance-strategies-for-rogues/

#2: Chain Mind Lightning Flay (?): This is a weird spell I don't see her cast too often, but it deals 600 damage per second to about eight people, so interrupt it.
#3: Curse of Blood: Increases Melee damage taken by 500.  Since she doesn't melee you, it only needs to be taken off of the main tank.
#4: Psychic Scream: Another spell I don't see her cast very often.  It's instant, and will fear you.

psychic scream / chain mind flay mention (chain mind lightning flay mmm interresting)

**Comment by kamicolo** [comment]
Additional note:
source: http://ru-mangos.ru/files/earlyServers/1.8.0/scripts/2006-04-19-Some%20Fixes%20for%20Boss%20And%20Instance%20Mob%20Spells.txt

namespace eval ::ai_HighPriestessJeklik {
	proc CanCast { npc victim } {
		set mobh [ ::GetHealthPCT $npc ]
		set spell_list "24437 23919 24673 12097 24687"
		;# Blood Leech, Swoop, Curse of Blood, Pierce Armor, Aspect of Jeklik
		set transform_spell_list "24209 23952 23953 15398 23966"
		;# Great Heal, Shadow Word: Pain, Mind Flay, Psychic Scream, Transform Troll
		set spellid [ lindex $spell_list [ expr { int( rand() * [ llength $spell_list ] ) } ] ]
		if { $mobh == 50 } {
			set spellid 23966
		} elseif { $mobh < 50 } {
			set spellid [ lindex $transform_spell_list [ expr { int( rand() * [ llength $transform_spell_list ] ) } ] ]
			if { $spellid == 24209 } { return [ ::AI::Cast $npc $npc $spellid ] }
		} else {
			set spellid [ lindex $spell_list [ expr { int( rand() * [ llength $spell_list ] ) } ] ]
		}
		return [ ::AI::Cast $npc $victim $spellid ]
	}
}

Blood leech mention / swoop / curse of blood / pierce armor / aspect of jeklik.

confirm Pierce Armor
confirm Psychic Scream
confirm Curse of Blood (However dunno which value the curse should be...)
https://www.youtube.com/watch?v=8yI57bjb2Yw

**Comment by kamicolo** [comment]
SPELL_SWOOP = 23919 // Confirmed
Notice in the youtube video a certain cleave effect is being done.

Sources:
http://ru-mangos.ru/files/earlyServers/1.8.0/scripts/2006-04-19-Some%20Fixes%20for%20Boss%20And%20Instance%20Mob%20Spells.txt
https://www.youtube.com/watch?v=8yI57bjb2Yw

**Comment by kamicolo** [comment]
SPELL_SUMMON_FRENZIED_BATS = 23974 // Confirmed HOWEVER,
Source: http://www.wowhead.com/spell=23974/summon-frenzied-bloodseeker-bats
Details are incorrect.
Instead a static summon value is used (8) this should be random between 6 - 8.

Sources:
http://wowwiki.wikia.com/wiki/High_Priestess_Jeklik
https://www.youtube.com/watch?v=8yI57bjb2Yw
https://www.youtube.com/watch?v=HulDb3F3rFw

**Comment by kamicolo** [comment]
SPELL_PIERCE_ARMOR = 12097 // MISSING 

Source: http://www.wowhead.com/spell=12097/pierce-armor
Details "might" be incorrect refer towards duration value.
Also think this should only occur in BAT FORM.

Notice in the move the main tank get pierce armor apllied.
Source: https://www.youtube.com/watch?v=8yI57bjb2Yw

**Comment by kamicolo** [comment]
SPELL_SHADOW_WORD_PAIN = 23952 // Confirmed however, (TROLL FORM ONLY)

Source: http://www.wowhead.com/spell=23952/shadow-word-pain
Details "might" be incorrect with values but, would be more a tweek.

**Comment by kamicolo** [comment]
SPELL_MIND_FLAY = 23953 // Confirmed

Source: http://www.wowhead.com/spell=23953/mind-flay
Details correct!

**Comment by kamicolo** [comment]
SPELL_BLOOD_LEECH = 22644 // Not certain HOWEVER, (BAT FORM ONLY)

Source: http://www.wowhead.com/spell=22644/blood-leech

Sources:
http://ru-mangos.ru/files/earlyServers/1.8.0/scripts/2006-04-19-Some%20Fixes%20for%20Boss%20And%20Instance%20Mob%20Spells.txt
It's being listed here. I stated bat form only because, it makes more sense in his bat form he vampiric leech blood from people since he has greater heal as troll form!

**Comment by kamicolo** [comment]
SPELL_GREATERHEAL = 23954 // Confirmed HOWEVER,

Source: http://www.wowhead.com/spell=23954/great-heal
Details are incorrect.

Several pages claim this spell should heal plus minus 30% HP value. 
Source: http://wow.gamepedia.com/High_Priestess_Jeklik_(tactics)
Healing amount versus HealthPoint Value 330,000 would be to weak.

**Comment by kamicolo** [comment]
GTG to work bummer..

curse of blood http://shendaar.pagesperso-orange.fr/GuideZG/Guide_ZG.htm
lalala will update when got time bye bye

**Comment by kamicolo** [comment]
Reserve note for badrider (bombing)

**Comment by kamicolo** [comment]
Reserve note summary.

Hey guy's just a quick word: I'm quite nostalgia with classic wow. Trying to assist getting back to the oldschool roots. Hope it's appreciated and you can work with it. Unfortunatly my code skills is well non existing however, gathering information or pointing towards the rite direction would be my support.

**Comment by kamicolo** [comment]
Mumble note.res. chain mind flay....

**Comment by killerwife** [comment]
@cala is this still relevant? If yes, explain how and if not close it pls.

**Comment by cala** [comment]
I don't think this issue was fully fixed. As such, it should probably remain open.

I say *probably* because I did not fully review it and don't have time to do so.

I'm still struggling to catch up with all the TBC-DB updates of the past months along with some changes in Viscidus/C'Thun scripts that don't show up in sniffs yet do happen, leading me to tear my hair apart...

P.S. : @killerwife I saw your notice to drop by in Discord, I'll try to do it soon.
